### ISSUE TITLE
Build failure x64/MSVC: Warning C4267 size_t to Uint  httplib

### ISSUE THREAD

**Issue by tfiner** [CesiumGS/cesium-native#251]
`C:\Work\lab\cesium-native\extern\cpp-httplib\httplib.h(2555): warning C4267: '=': conversion from 'size_t' to 'uInt', possible loss of data [C:\Work\lab\cesium-native\build\CesiumIonClient\CesiumIonClient.vcxproj]`

Followed by 
`C:\Work\lab\cesium-native\extern\cpp-httplib\httplib.h(2555): error C2220: warning treated as error - no 'object' file generated [C:\Work\lab\cesium-native\build\CesiumIonClient\CesiumIonClient.vcxproj]`

The solution is simple for x64, static_cast size_t to uInt.  Although, if anyone ever compresses more than 4gb, you'll probably want to revisit this by either using a safe_cast that throws on overflow, or breaking the data up into chunks.

I recommend turning warnings as errors off until the code is more stable and you have all your supported compilers / platforms being used by multiple people to catch problems like this.

**Comment by kring** [comment]
@tfiner VS2017 on Windows x64 is our primary compilation target for cesium-native, and we haven't seen that warning. Are you building cesium-native has part of another project that overrides the compilation settings perhaps? Generally, third-party headers like this should be considered "system headers" and get less strict warning treatment.

**Comment by kring** [comment]
Here's the log of our CI system building the latest main (as of this writing) on Windows x64 with VS2017:
https://travis-ci.com/github/CesiumGS/cesium-native/jobs/506616305

It's also built with VS2019 and for Clang and GCC on Linux:
https://github.com/CesiumGS/cesium-native/runs/2629831808

**Comment by tfiner** [comment]
I'm actually building cesium-native by itself with a fairly up to date VS 2019.  A couple of caveats:
1. That warning is legitimate: if you assign a 32 bit unsigned int with a size_t on x64, it is truncating the integer.  Now, is it a problem really?  Probably not.  
2. I didn't upgrade the toolset. My understanding is that 2019 uses 2017 under the hood when you don't upgrade the toolset.  But.. this is probably the reason for my seeing the warning that you guys aren't.

I agree with you about third party headers, they should be looser and should probably have warnings as errors turned off.  

Otherwise, you'll run into more 3rd party libraries that prevent builds like this one did.  

Also, you might want to include 2019 in your builds, it's been out a few years and is stable and ABI compatible.  I understand holding off though, I worked at a place where the upgrade to 2017 was particularly painful.

**Comment by kring** [comment]
Yeah we build every commit and branch against VS2019, and it's clean there too. I can't really explain why you're seeing warnings in that header and we're not.

**Comment by kring** [comment]
Just to be clear: I appreciate it's a legitimate warning, not a false positive. But the higher warning level and "treat warnings as errors" behavior used in cesium-native shouldn't apply to "system" headers like httplib.h. So the warning probably shouldn't be displayed, and definitely shouldn't cause the build to fail. That's consistent with what I see on my system and on our CI build machine. Are you running the latest patch version of VS2017 and 2019?

**Comment by tfiner** [comment]
Weird!  Yeah, I'm building using the latest

**Comment by tfiner** [comment]
Dang it, I closed this by accident.

**Comment by tfiner** [comment]
Thank you.  Here are my versions:
Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27045 for x64
Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29915 for x64

cmake version 3.20.2
cpp-httplib is at commit cf475bcb50567

Maybe there is some interaction with CMake?  I saw this when I followed the instructions for building this in the readme (create the VS solution using CMake, then build the solution using VS).  In a clean directory, I cloned cesium-native from within VS and used its CMake integration.  I didn't see the warning / error there.

I have noticed in previous times new warnings appearing when headers are different too.  Maybe the difference is in the windows SDK selected by CMake?  The version that CMake / VS picked was 10.0.18362.0.
